### PR TITLE
`Development`: Enhance exam creation e2e tests

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

During the creation of an exam we currenlty did only check some fields on the exam details page. We did not check the returned exam object for correct values.

### Description

This PR adds more checks to the details page and also checks the returned exam creation object more extensivly. 

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test

### CODE REVIEW

**Review by rriyaldhi — DISMISSED** [review]
Code changes look good.

**Review by JohannesStoehr — DISMISSED** [review]
Refactoring and new assertions look good to me

File: src/test/cypress/e2e/exam/ExamAssessment.cy.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ describe('Exam assessment', () => {
         });
 
         beforeEach('Create exam, exercise and submission', () => {
+            cy.login(admin);
```
Comment (krusche) [inline src/test/cypress/e2e/exam/ExamAssessment.cy.ts:74]: why is an admin login necessary here?
Creating exercises should done by instructors

File: src/test/cypress/e2e/exam/ExamAssessment.cy.ts [hunk 2]
```diff
@@ -117,6 +118,7 @@ describe('Exam assessment', () => {
 
         describe('Modeling exercise assessment', () => {
             beforeEach('Create exercise and submission', () => {
+                cy.login(admin);
```
Comment (krusche) [inline src/test/cypress/e2e/exam/ExamAssessment.cy.ts:121]: why is an admin login necessary here?
Creating exercises should done by instructors

File: src/test/cypress/e2e/exam/ExamAssessment.cy.ts [hunk 3]
```diff
@@ -151,6 +153,7 @@ describe('Exam assessment', () => {
 
         describe('Text exercise assessment', () => {
             beforeEach('Create exercise and submission', () => {
+                cy.login(admin);
```
Comment (krusche) [inline src/test/cypress/e2e/exam/ExamAssessment.cy.ts:156]: why is an admin login necessary here?
Creating exercises should done by instructors

File: src/test/cypress/e2e/exam/ExamAssessment.cy.ts [hunk 4]
```diff
@@ -190,6 +193,7 @@ describe('Exam assessment', () => {
         });
 
         beforeEach('Create exercise and submission', () => {
+            cy.login(admin);
```
Comment (krusche) [inline src/test/cypress/e2e/exam/ExamAssessment.cy.ts:196]: why is an admin login necessary here?
Creating exercises should done by instructors

File: src/test/cypress/e2e/exam/ExamAssessment.cy.ts (hunk 1 above)
Comment (TheZoker) [reply src/test/cypress/e2e/exam/ExamAssessment.cy.ts]: In this case the admin account is used to authenticate the API call to create a exercises. The exercise it not created via the UI here. But I agree that we should also use a instructor account for this call. Fixed in the latest commit

**Pull request merged**